GeneralsMD/Code/GameEngine/Source/GameClient/GlobalLanguage.cpp

```diff
@@ -191,10 +191,11 @@ void GlobalLanguage::parseFontFileName( INI *ini, void * instance, void *store,
 
 Int GlobalLanguage::adjustFontSize(Int theFontSize)
 {
-	Real adjustFactor = TheGlobalData->m_xResolution / (Real)DEFAULT_DISPLAY_WIDTH;
-	adjustFactor = 1.0f + (adjustFactor-1.0f) * m_resolutionFontSizeAdjustment;
+	// TheSuperHackers @bugfix Mauller 25/08/2025 Scale fonts based on the smallest screen dimension so they are independent of aspect ratio
+	// We also rescale the font size adjustment to make the ingame fonts closer to what players have been used to with the original scaling
+	Real adjustFactor = min(TheGlobalData->m_xResolution / (Real)DEFAULT_DISPLAY_WIDTH, TheGlobalData->m_yResolution/ (Real)DEFAULT_DISPLAY_HEIGHT );
+	adjustFactor = 1.0f + (adjustFactor-1.0f) * ( 0.25f + m_resolutionFontSizeAdjustment );
```

Comment:
0.25f is a magic value that does not belong here. It will also change the scaling in 1600 x 1200 (4:3). I do not like this solution, because it changes the promise of `m_resolutionFontSizeAdjustment`.

Reply:
It would be better to use the default value of the new scaler in the Options.ini to apply the new adjustment.